awx/ui_next/src/screens/Inventory/InventoryDetail/InventoryDetail.jsx

```diff
@@ -11,37 +11,28 @@ import DeleteButton from '@components/DeleteButton';
 import ContentError from '@components/ContentError';
 import ContentLoading from '@components/ContentLoading';
 import { InventoriesAPI } from '@api';
+import useRequest from '@util/useRequest';
 import { Inventory } from '../../../types';
 
 function InventoryDetail({ inventory, i18n }) {
-  const [instanceGroups, setInstanceGroups] = useState([]);
-  const [hasContentLoading, setHasContentLoading] = useState(true);
-  const [contentError, setContentError] = useState(null);
   const history = useHistory();
-  const isMounted = useRef(null);
+
+  const {
+    result: instanceGroups,
+    isLoading,
+    error,
+    request: fetchInstanceGroups,
+  } = useRequest(
+    useCallback(async () => {
```

Comment:
When _wouldn't_ we wrap the promise in `useCallback` before passing it into the helper? Can or should we also put that in the hook?

Reply:
It isn't necessary if the function isn't defined inline (i.e. it's defined outside the component or something). The problem with moving it into the hook is we would also have to pass in the dependencies array (`[inventory.id]`), and once it's a function parameter, the linter can no longer statically verify it includes everything the function references.